controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -128,18 +132,22 @@ func (r *AKODeploymentConfigReconciler) reconcileAVIDelete(
 		return res, err
 	}
 
-	return phases.ReconcileClustersPhases(ctx, r.Client, log, obj,
-		// When AKODeploymentConfig is being deleted, reconcile avi user
-		// deletion no matter cluster is being deleted or not
-		[]phases.ReconcileClusterPhase{
-			r.userReconciler.ReconcileAviUserDelete,
-		},
-		[]phases.ReconcileClusterPhase{
-			// TODO(fangyuanl): handle the data network configuration
-			// deletion
-			r.userReconciler.ReconcileAviUserDelete,
+	return phases.ReconcilePhases(ctx, log, obj, []phases.ReconcilePhase{
+		r.reconcileAviInfraSettingDelete,
```

Comment:
why this `reconcileAviInfraSettingDelete` not inside `ReconcileClustersPhases`?

Reply:
is it possible for it to be aligned with other phases below such as `ReconcileAviUserDelete`

Reply:
reconcileAviInfraSettingDelete() should be triggered when `len(clusters.Items) == 0`. So it won't be triggered inside ReconcileClustersPhases